ee/agent/knapsack/knapsack.go

```diff
@@ -239,3 +242,42 @@ func (k *knapsack) CurrentEnrollmentStatus() (types.EnrollmentStatus, error) {
 
 	return types.Enrolled, nil
 }
+
+//Todo change where the enrollment details call is made
+// consider goroutine to unblock the startup of launcher
+// we can use the latest osquery version stored in the knapsack to add to the enrollment details when doing autoupdate
+
+func (k *knapsack) SetEnrollmentDetails(details types.EnrollmentDetails) error {
```

Comment:
We aren't ever returning an error here -- I think you could update this signature to not have a return